src/posix/sdl/hardware.cpp

```diff
@@ -334,8 +334,8 @@ CUSTOM_CVAR(Bool, swtruecolor, true, CVAR_ARCHIVE|CVAR_GLOBALCONFIG|CVAR_NOINITC
 
 CUSTOM_CVAR (Bool, fullscreen, false, CVAR_ARCHIVE|CVAR_GLOBALCONFIG)
 {
-	NewWidth = screen->VideoWidth;
-	NewHeight = screen->VideoHeight;
+	NewWidth = screen->GetWidth();
```

Comment:
I cannot accept this change, here. With the new scaling feature, this will cause problems.